File: api/src/endpoint/organization/list_organizations.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::OrganizationResource};

pub struct ListOrganizations;

impl Endpoint for ListOrganizations {
    const PATH: &'static str = "/organizations";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListOrganizationsRequest;
    type Response = ListOrganizationsResponse;
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ListOrganizationsRequest {}

pub type ListOrganizationsResponse = Vec<OrganizationResource>;

File: gitdot-api-ts/src/endpoint/organization/list-organizations.ts
import { z } from "zod";

import { OrganizationResource } from "../../resource";

export const ListOrganizationsRequest = z.object({});
export type ListOrganizationsRequest = z.infer<typeof ListOrganizationsRequest>;

export const ListOrganizations = {
  path: "/organizations",
  method: "GET",
  request: ListOrganizationsRequest,
  response: z.array(OrganizationResource),
} as const;
export type ListOrganizations = typeof ListOrganizations;
